tests/test_multiline_variables.py

```diff
@@ -3,47 +3,51 @@
 import os
 import subprocess
 
-import pytest
-
-from universum.lib.module_arguments import ModuleArgumentParser, IncorrectParameterError
-from universum.lib.utils import read_and_check_multiline_option
+from universum.lib.gravity import construct_component
+from universum.lib.module_arguments import ModuleArgumentParser
+from universum.modules.error_state import HasErrorState
 
 text = "This is text\nwith some line breaks\n"
 error_message = "This is some missing argument error message"
 
 script = f"""
 from universum.lib.module_arguments import ModuleArgumentParser
-from universum.lib.utils import read_and_check_multiline_option
+from universum.lib.gravity import construct_component
+from universum.modules.error_state import HasErrorState
 
 parser = ModuleArgumentParser()
-parser.add_argument('--argument', '-a', metavar='ARGUMENT')
+parser.add_argument('--argument', '-a', dest='HasErrorState.argument', metavar='ARGUMENT')
 
 namespace = parser.parse_args()
-print(read_and_check_multiline_option(namespace, 'argument', "{error_message}"))
+module = construct_component(HasErrorState, namespace)
+print(module.read_and_check_multiline_option('argument', "{error_message}"))
+if module.is_in_error_state():
+    print(module.global_error_state.get_errors()[0])
 """
 
 
-@pytest.fixture()
-def parser():
-    parser = ModuleArgumentParser()
-    parser.add_argument('--argument', '-a', metavar='ARGUMENT')
-    yield parser
+def construct_test_module(args):
```

Comment:
Why is function better than fixture?

Reply:
Fixture is only needed when there is something to tear down.
Function is easier to navigate and debug in IDE .